code/game/objects/structures/transit_tubes/station.dm

```diff
@@ -27,116 +29,128 @@
 
 // Stations which will send the tube in the opposite direction after their stop.
 /obj/structure/transit_tube/station/reverse
-	reverse_launch = 1
+	reverse_launch = TRUE
 
 /obj/structure/transit_tube/station/should_stop_pod(pod, from_dir)
-	return 1
+	return TRUE
 
-/obj/structure/transit_tube/station/Bumped(mob/AM as mob|obj)
-	if(!pod_moving && icon_state == "open" && istype(AM, /mob/living) && !is_type_in_list(AM, disallowed_mobs))
+/obj/structure/transit_tube/station/Bumped(mob/living/L)
+	if(!pod_moving && hatch_state == TRANSIT_TUBE_OPEN && isliving(L) && !is_type_in_list(L, disallowed_mobs))
+		var/failed = FALSE
 		for(var/obj/structure/transit_tube_pod/pod in loc)
 			if(pod.contents.len)
-				to_chat(AM, "<span class=The pod is already occupied.</span>")
-				return
+				failed = TRUE
 			else if(!pod.moving && (pod.dir in directions()))
-				AM.forceMove(pod)
+				pod.move_into(L)
 				return
+		if(failed)
+			to_chat(L, "<span class='warning'>The pod is already occupied.</span>")
 
 
 
 /obj/structure/transit_tube/station/attack_hand(mob/user as mob)
```

Comment:
The formatting of the logic in this proc is SO disgusting and hard to read, it works but.. bonus points if you refactor it with early returns, continues, more multi conditional ifs etc.

Reply:
Yep ugly shit. Refactored that as well